### ISSUE TITLE
Imapcsync: cancel form is broken

### ISSUE THREAD

**Issue by stephdl** [NethServer/dev#7052]
When a task is created, it's validated and added correctly to the list. However, when you edit an existing task and decide to cancel the changes, the list of tasks still reflects the changes made during the edit, even though the cancel action should revert those changes.

This suggests that the task list might be getting updated prematurely (during the editing process) rather than after confirming the changes. The input values from the edit phase are likely being applied to the task list in real-time instead of being committed only after validation (when the user confirms the edit). To fix this, the application should ensure that the task list remains unchanged until the "save" or "confirm" action is triggered, and canceling should discard any temporary edits.

https://github.com/user-attachments/assets/553addbb-86a7-490b-8f06-e4afeeefe1d5

**Steps to reproduce**

- Install an account provider, a mail server, and imapsync.
- Configure a task.
- The task is validated and successfully saved.
- The task appears correctly in the list view.
- Edit the task.
- Enter invalid values into the input fields.
- Click the "Cancel" button.
- Despite canceling, the task in the table is still updated with the incorrect values.

**Expected behavior**

I expect the tasks in the table to remain unchanged because the edit process was not completed and was canceled before any changes were confirmed.
**Actual behavior**

The behavior is that the task displayed in the database view does not match what is shown after refreshing the page.

**Components**

imapsync 1.0.3

**Comment by stephdl** [comment]
QA 

target version is ghcr.io/nethserver/imapsync:1.0.4-dev.2

- install the version of testing in a terminal
`add-module ghcr.io/nethserver/imapsync:1.0.4-dev.2`
or update a module
`api-cli run update-module --data '{"module_url":"ghcr.io/nethserver/imapsync:1.0.4-dev.2","instances":["imapsync1"],"force":true}'`

- you have to verify the bug is not reproducible
test 1
- create a task without saving, hit the cancel button
- when you create a new task the content is empty from the previous task 
test 2
- create and save a task
- open the previous task and modify the input like remote username, remote host, password, .... without saving and hit the button cancel
- the table of tasks must not have modified like in the video of the bug description, the remote username, the remote host must be the same, if you refresh the task with the button, the content must be the same as before

**Comment by stephdl** [comment]
community verification is asked to : https://community.nethserver.org/t/in-imapsync-it-fails-to-show-me-tasks/24572/19

**Comment by stephdl** [comment]
verified at https://community.nethserver.org/t/in-imapsync-it-fails-to-show-me-tasks/24572/26
